### PR TITLE
check4Stata: Add first helper functions

### PR DESCRIPTION
**Pull request by weigtnik**
This PR addresses #110, adding two helper functions that the main/wrapper function will use.
- `checkIntOverflow` is used to detect labeled (whole number/integer) values that exceed the highest integer R can work with, which would cause problems when exporting to `.dta` because Stata only allows for labeled *integers*.
- `checkLabeledFractionals` addresses said restriction of Stata by looking for labeled fractional values (or fractional values tagged as missing).

This is not urgent, and I will be happy about all the feedback you can provide!

### CODE REVIEW

File: R/checkIntOverflow.R [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#' Check a \code{GADSdat} for large integers.
+#'
+#' Check a \code{GADSdat} object for any occurrences of integer values, esp.
+#' in the metadata, that are too large for R to handle.
+#' in its meta data which could cause problems esp. when exporting into \code{.dta} format.
```
Comment (beckerbenj) [inline R/checkIntOverflow.R:5]: Is this a duplicated/alternative line?

File: tests/testthat/test_checkIntOverflow.R [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  df2 <- changeMissings(df2, "VAR1", c(-9999999999, 9999999999), c("miss", "valid"))
+  check_result <- checkIntOverflow(df2)
+  outlist2 <- outlist
+  outlist2[1,] <- list("VAR1", -9999999999, "miss", TRUE, 1)
+  outlist2[2,] <- list("VAR1", 9999999999, "valid", TRUE, 6)
+  expect_equal(check_result, outlist2)
+})
+
+# remove this last one?
+test_that("Two very large labeled values in the same variable where one value is empty", {
```
Comment (beckerbenj) [inline tests/testthat/test_checkIntOverflow.R:37]: Maybe instead you could add a test where there are no "empty" value labels at all in a variable?

File: tests/testthat/test_checkLabeledFractionals.R [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+dfSAV <- import_spss(file = test_path("helper_spss_missings.sav"))
+df1 <- changeValLabels(dfSAV, "VAR1", 2, "Apple")
+
+# general output structure to be changed for each test
+outlist <- data.frame(varName = "<none found>",
+                      value = NA_real_,
+                      missings = NA_character_,
+                      empty = NA)
+
+test_that("Correctly identify metadata without labeled fractional values", {
```
Comment (beckerbenj) [inline tests/testthat/test_checkLabeledFractionals.R:11]: LGTM

File: tests/testthat/test_checkLabeledFractionals.R [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  check_result <- checkLabeledFractionals(df1)
+  expect_equal(check_result, outlist)
+})
+
+test_that("One existing labeled fractional in one variable", {
```
Comment (beckerbenj) [inline tests/testthat/test_checkLabeledFractionals.R:16]: LGTM

**Comment by weigtnik** [comment]
Thanks for your thorough review and your comments! I will try to address them over the course of this week. 🙌

**Comment by weigtnik** [comment]
Regarding your (somewhat) major/general requests:

1. I absolutely agree, and in hindsight, I don't even understand why I left out the argument names in the tests.
2. Thanks for bringing up the output/return format! The first option you described sounds preferable to me, too. I have now been thinking about a `verbose` option to get a message saying "all good". But I don't think that's a very useful addition to the function.

File: R/checkIntOverflow.R [hunk 5]
```diff
@@ -0,0 +1,73 @@
+#' Check a \code{GADSdat} for large integers.
+#'
+#' Check a \code{GADSdat} object for any occurrences of integer values, esp.
```
[the remark this answers is not in the document]
Comment (weigtnik) [reply R/checkIntOverflow.R:3]: Thanks for your comment and your input! I have changed the wording to "whole-number values" for the "`GADSdat` side" of affairs, and also tried to explain the issue a bit more. I'm looking forward to your feedback on the changes!

Also, thanks for spotting the oversight regarding large floating-point/fractional values! I have fixed that and added a corresponding test.

File: R/checkLabeledFractionals.R [hunk 6]
```diff
@@ -0,0 +1,58 @@
+#' Check meta data for fractional values.
+#'
+#' Check a \code{GADSdat} object for any occurrences of fractional values
+#' in its meta data which could cause problems esp. when exporting into \code{.dta} format.
```
[the remark this answers is not in the document]
Comment (weigtnik) [reply R/checkLabeledFractionals.R:4]: I thought that the description would then be too redundant with the title of the page. But I think I get why this (in terms of manual page sections) is more of a detail than a general description - and especially why a more detailed elaboration would be in order.

[the remark this answers is not in the document]
Comment (weigtnik) [reply R/checkLabeledFractionals.R:4]: Thanks for the resource! I think "[just] Try to find a slightly different wording" is a good instruction for me. 👍

File: R/checkIntOverflow.R [hunk 7]
```diff
@@ -33,40 +35,34 @@ checkIntOverflow <- function(GADSdat) {
...
                     value = NA_real_,
                     missings = NA_character_,
                     empty = NA,
-                    rownum = NA_integer_)
+                    rownum = NA_integer_)[0,]
 
-  # option 1
-  # huge_number_rows <- which((labels$labeled == "yes") &
-  #                            (labels$value > 2147483647 | labels$value < -2147483647))
+  huge_labeled_number_rows <- which(abs(labels$value) > .Machine$integer.max &
```
Comment (beckerbenj) [inline R/checkIntOverflow.R:44]: LGTM!

File: R/checkIntOverflow.R [hunk 8]
```diff
@@ -33,40 +35,34 @@ checkIntOverflow <- function(GADSdat) {
   check_GADSdat(GADSdat)
   labels <- GADSdat$labels
 
-  out <- data.frame(varName = "<none found>",
+  out <- data.frame(varName = NA_character_,
```
Comment (beckerbenj) [inline R/checkIntOverflow.R:38]: Nit:

Could you change this to use empty vectors in the first place? 
```
data.frame(varName = character(),
                   value = numeric(), ....)
```
and omit the subsetting `[0,]`

File: R/checkLabeledFractionals.R [hunk 9]
```diff
@@ -27,31 +29,31 @@ checkLabeledFractionals <- function(GADSdat) {
   check_GADSdat(GADSdat)
   labels <- GADSdat$labels
 
-  out <- data.frame(varName = "<none found>",
+  out <- data.frame(varName = NA_character_,
```
Comment (beckerbenj) [inline R/checkLabeledFractionals.R:32]: See above

File: tests/testthat/test_checkIntOverflow.R [hunk 10]
```diff
@@ -1,45 +1,73 @@
...
 dfSAV <- import_spss(file = test_path("helper_spss_missings.sav"))
-df1 <- changeValLabels(dfSAV, "VAR1", 2, "Apple")
+df1 <- changeValLabels(GADSdat = dfSAV,
+                       varName = "VAR1",
+                       value = 2,
+                       valLabel = "Apple")
 
 # general output structure to be changed for each test
-outlist <- data.frame(varName = "<none found>",
+outlist <- data.frame(varName = NA_character_,
```
Comment (beckerbenj) [inline tests/testthat/test_checkIntOverflow.R:9]: see above

**Review by beckerbenj — APPROVED** [review]
Thanks for the changes, looks all good to me!

Some minor nits in the respective comments. 

Feel free to merge (and delete the branch).
